llvm/tools/llvm-objdump/llvm-objdump.cpp

```diff
@@ -2051,41 +2051,51 @@ disassembleObject(ObjectFile &Obj, const ObjectFile &DbgObj,
+          // This symbol didn't trigger any interesting handling. Try the other
+          // symbols defined at this address. Errors can only be returned if the
+          // disassembler responds.
+          cantFail(std::move(Err));
           continue;
         }
 
-        if (*Status == MCDisassembler::Fail) {
-          // If onSymbolStart returns Fail, that means it identified some kind
-          // of special data at this address, but wasn't able to disassemble it
-          // meaningfully. So we fall back to disassembling the failed region
-          // as bytes, assuming that the target detected the failure before
-          // printing anything.
-          //
-          // Return values Success or SoftFail (i.e no 'real' failure) are
-          // expected to mean that the target has emitted its own output.
-          //
-          // Either way, 'Size' will have been set to the amount of data
-          // covered by whatever prologue the target identified. So we advance
-          // our own position to beyond that. Sometimes that will be the entire
-          // distance to the next symbol, and sometimes it will be just a
-          // prologue and we should start disassembling instructions from where
-          // it left off.
-          outs() << DT->Context->getAsmInfo()->getCommentString()
-                 << " error in decoding " << SymNamesHere[SHI]
-                 << " : decoding failed region as bytes.\n";
-          for (uint64_t I = 0; I < Size; ++I) {
-            outs() << "\t.byte\t " << format_hex(Bytes[I], 1, /*Upper=*/true)
-                   << "\n";
+        // If onSymbolStart responded and set Err, that means it identified some
+        // kind of special data at this address, but wasn't able to disassemble
+        // it meaningfully. So we fall back to printing the error out and
+        // disassembling the failed region as bytes, assuming that the target
+        // detected the failure before printing anything.
+        //
+        // Either way, 'Size' will have been set to the amount of data covered
+        // by whatever prologue the target identified. So we advance our own
+        // position to beyond that. Sometimes that will be the entire distance
+        // to the next symbol, and sometimes it will be just a prologue and we
+        // should start disassembling instructions from where it left off.
+
+        if (Err) {
+          std::string ErrMsgStr = toString(std::move(Err));
+          StringRef ErrMsg = ErrMsgStr;
+          do {
+            StringRef Line;
+            std::tie(Line, ErrMsg) = ErrMsg.split('\n');
+            outs() << DT->Context->getAsmInfo()->getCommentString()
+                   << " error decoding " << SymNamesHere[SHI] << ": " << Line
+                   << '\n';
+          } while (!ErrMsg.empty());
+
+          if (Size) {
+            outs() << DT->Context->getAsmInfo()->getCommentString()
+                   << " decoding failed region as bytes.\n";
```

Comment:
Nit: I know this isn't strictly an error message, but it's similar to one, so should probably follow the corresponding coding standard, namely that it shouldn't have a "." at the end. OTOH, if this is the style used elsewhere (I haven't looked) for this sort of output, it's fine to leave it.

Reply:
It doesn't seem to be used elsewhere, so I removed it in the update.